test/blackbox-tests/test-cases/env/env-bin-pform.t/run.t

```diff
@@ -1,11 +1,11 @@
 %{exe:foo.exe} should not be veisible if foo.exe is added to PATH via the
 binaries stanza. %{bin:foo} is visible on the other hand.
   $ dune build
+           foo alias default
+  this is foo.exe
```

Comment:
I don't understand why this changed and if this change is desirable. Do we delay all errors until the end of the build now?

Reply:
Now if a memoized async function raises, we'll only re-raise after it finishes executing. It's probably slightly worse, but ultimately this is the consistent behavior.

Reply:
Right, I think I understand.

Just to double-check: if we execute a compound computation `A | B` and `A` raises, previously we would report the error immediately, and now only after `B` finishes (raises or completes successfully)?

Reply:
> if we execute a compound computation A | B and A raises, previously we would report the error immediately, and now only after B finishes (raises or completes successfully)?

You're describing the current behavior correctly, but I think the way it worked before was to only eagerly raise the *first* error. Other errors were discarded as you can see from the test.

Reply:
I'll leave a comment in the code about this. It's worth re-visiting this later.

Reply:
Thanks! Yes, adding a comment about this would be great.